validator/bigquery_cmek.rego

```diff
@@ -0,0 +1,42 @@
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPBigqueryCMEKEncryptionConstraintV1
+
+import data.validator.gcp.lib as lib
+
+###########################
+# Find Whitelist Violations
+###########################
+deny[{
+	"msg": message,
+	"details": metadata,
+}] {
+	asset := input.asset
+	asset.asset_type == "bigquery.googleapis.com/Table"
+
+	# Find KMS key name for the table
+	kmskeyname := lib.get_default(asset.resource.data, "encryptionConfiguration", "NOTFOUND")
+
+	# Check if KMS is enabled
+	kmskeyname == "NOTFOUND"
+
+	message := sprintf("%v does not have the client managed encryption key setup.", [asset.name])
+	metadata := {
+		"resource": asset.name,
+		"type": "encryptionerror",
```

Comment:
curious: where does the type come from? Is there a canonical list somewhere that we can reference?

Reply:
The type is just a string. We don't have any canonical list from where we pull this. I just made it up.

Reply:
is it useful to have this made up type then? isn't that inferred by the constraint? how do we envision this type to be used? I'd suggest removing this unless we have a concrete use case.